src/hotspot/share/gc/shared/stringdedup/stringDedupTable.cpp

```diff
@@ -421,8 +421,10 @@ size_t StringDedup::Table::_number_of_entries = 0;
 size_t StringDedup::Table::_grow_threshold;
 StringDedup::Table::CleanupState* StringDedup::Table::_cleanup_state = nullptr;
 bool StringDedup::Table::_need_bucket_shrinking = false;
-volatile size_t StringDedup::Table::_dead_count = 0;
-volatile StringDedup::Table::DeadState StringDedup::Table::_dead_state = DeadState::good;
+Atomic<size_t> StringDedup::Table::_dead_count{};
```

Comment:
```suggestion
Atomic<size_t> StringDedup::Table::_dead_count;
```
This is not a function, is it?

Reply:
With the braces is a value-initialization of the variable. Without is default initialization. As it happens,
because this is a static storage duration variable and the class has a default ctor, those are the same.
But I'd prefer to be explicit.